### PR TITLE
MTL-1981 - Document operational procedure for configuring NCNs with bonded HSN connections

### PR DESCRIPTION
**Pull request by spillerc-hpe**
# Description

User documentation for the `csm.ncn.hsn_bonding` Ansible role for configuring HSN bonding on NCN worker nodes.

# Checklist

- [X] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [X] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [X] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Comment by mtupitsyn** [comment]
@spillerc-hpe In order to get this merged for upcoming release, please acquire sufficient amount of approvals. @nrockershousen could you please take a look?

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+* More than one bonded interface per NCN worker node is not supported.
+
+## Prerequisites
+
+The following steps should have occurred before configuring a bonded interface on a NCN worker node.
+
+* The Slingshot Fabric Manager is installed and configured.
+* The Slingshot Host Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* The User Services Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* Link Aggregation Groups (LAG) have been created using the Slingshot Fabric Manager.
```
Comment (manderson-hpe) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]: Could remove `the` here?

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+* [How to create a bonded IP host interface with HPE Slingshot](https://support.hpe.com/hpesc/public/docDisplay?docId=dp00004881en_us&docLocale=en_US)
+
+## Limitations
+
+* This procedure is only supported on nodes with HPE Cassini interfaces.
+* More than one bonded interface per NCN worker node is not supported.
+
+## Prerequisites
+
+The following steps should have occurred before configuring a bonded interface on a NCN worker node.
```
Comment (manderson-hpe) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]: `a NCN` to `an NCN` for consistency in the document.

**Review by manderson-hpe — APPROVED** [review]
Approved with a few minor changes.

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md (hunk 1 above)
Comment (spillerc-hpe) [reply operations/configuration_management/Configure_HSN_NIC_Bonding.md]: Fixed

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md (hunk 2 above)
Comment (spillerc-hpe) [reply operations/configuration_management/Configure_HSN_NIC_Bonding.md]: Fixed

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+    * [Additional steps](#additional-steps)
+
+This procedure can be used to create a bonded HSN interface on an NCN worker node. The `csm.ncn.hsn_bonding` Ansible role is
+merely an automation of the manual steps outlined in the "How to create a bonded IP host interface with HPE Slingshot" document (See [References](#references))
+
+## References
+
+The "How to create a bonded IP host interface with HPE Slingshot" document is available from the HPE Support Portal. The other documentation is bundled with the HPE Slingshot software download.
+
+* HPE Slingshot Installation Guide for CSM
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* _HPE Slingshot Installation Guide for CSM_
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+* HPE Slingshot Administration Guide
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* _HPE Slingshot Administration Guide_
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+    * [Verification](#verification)
+    * [Troubleshooting](#troubleshooting)
+    * [Additional steps](#additional-steps)
+
+This procedure can be used to create a bonded HSN interface on an NCN worker node. The `csm.ncn.hsn_bonding` Ansible role is
+merely an automation of the manual steps outlined in the "How to create a bonded IP host interface with HPE Slingshot" document (See [References](#references))
+
+## References
+
+The "How to create a bonded IP host interface with HPE Slingshot" document is available from the HPE Support Portal. The other documentation is bundled with the HPE Slingshot software download.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
The _How to create a bonded IP host interface with HPE Slingshot_ document is available from the HPE Support Portal. The other documentation is bundled with the HPE Slingshot software download.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+
+* HPE Slingshot Installation Guide for CSM
+* HPE Slingshot Administration Guide
+* [How to create a bonded IP host interface with HPE Slingshot](https://support.hpe.com/hpesc/public/docDisplay?docId=dp00004881en_us&docLocale=en_US)
+
+## Limitations
+
+* This procedure is only supported on nodes with HPE Cassini interfaces.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* This procedure is only supported on nodes with HPE Slingshot 200Gbps interfaces.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+* More than one bonded interface per NCN worker node is not supported.
+
+## Prerequisites
+
+The following steps should have occurred before configuring a bonded interface on an NCN worker node.
+
+* Slingshot Fabric Manager is installed and configured.
+* Slingshot Host Software is installed and an image containing it has been deployed to the NCN worker nodes.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* Slingshot Host Software (SHS) is installed and an image containing it has been deployed to the NCN worker nodes.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+* User Services Software is installed and an image containing it has been deployed to the NCN worker nodes.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* User Services Software (USS) is installed and an image containing it has been deployed to the NCN worker nodes.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+## Limitations
+
+* This procedure is only supported on nodes with HPE Cassini interfaces.
+* More than one bonded interface per NCN worker node is not supported.
+
+## Prerequisites
+
+The following steps should have occurred before configuring a bonded interface on an NCN worker node.
+
+* Slingshot Fabric Manager is installed and configured.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* HPE Slingshot Fabric Manager is installed and configured.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+* Slingshot Host Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* User Services Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* Link Aggregation Groups (LAG) have been created using the Slingshot Fabric Manager.
+    * An IP address and netmask have been provided by the fabric administrator.
+    * The bonding mode used for the LAG has been provided by the fabric administrator.
+    * The DMAC used for the LAG has been provided by the fabric administrator (See "How to create a bonded IP host interface with HPE Slingshot") for more information.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
    * The DMAC used for the LAG has been provided by the fabric administrator (See the _How to create a bonded IP host interface with HPE Slingshot_ document) for more information.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+* More than one bonded interface per NCN worker node is not supported.
+
+## Prerequisites
+
+The following steps should have occurred before configuring a bonded interface on an NCN worker node.
+
+* Slingshot Fabric Manager is installed and configured.
+* Slingshot Host Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* User Services Software is installed and an image containing it has been deployed to the NCN worker nodes.
+* Link Aggregation Groups (LAG) have been created using the Slingshot Fabric Manager.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
* Link Aggregation Groups (LAG) have been created using the HPE Slingshot Fabric Manager.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 12]
```diff
@@ -0,0 +1,418 @@
...
+    * The bonding mode used for the LAG has been provided by the fabric administrator.
+    * The DMAC used for the LAG has been provided by the fabric administrator (See "How to create a bonded IP host interface with HPE Slingshot") for more information.
+
+## Setup
+
+### Fabric configuration
+
+Configuring a LAG using the Slingshot Fabric Manager is beyond the scope of this document (See the Link Aggregation section of the
+"HPE Slingshot Installation Guide for CSM" for more information) however the following example configuration is provided for the
+purpose of illustration.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
Configuring a LAG using the HPE Slingshot Fabric Manager is beyond the scope of this document (See the "Link Aggregation" section of the
_HPE Slingshot Installation Guide for CSM_ for more information), however the following example configuration is provided for the
purpose of illustration.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 13]
```diff
@@ -0,0 +1,418 @@
...
+      lastUpdated = "2024-10-11T11:13:39Z"
+      name = "hsn-nic-bonding"
+      [[layers]]
+      cloneUrl = "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
+      commit = "e8de8f5be0ba6576b4102137db821a0da3b28375"
+      name = "hsn-nic-bonding"
+      playbook = "ncn_hsn_bonding.yml"
+      ```
+
+1. Create a CFS session to apply the configuration to the node(s).
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md:210]: Please indent everything under this step.

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 14]
```diff
@@ -0,0 +1,418 @@
...
+          port number: 3
+          port state: 63
+      ```
+
+## Troubleshooting
+
+Refer to [View Configuration Session Logs](./View_Configuration_Session_Logs.md) to troubleshoot why the CFS session failed to complete successfully.
+
+If the underlying HSN interfaces are not up or present refer to the Slingshot documentation listed in [References](#references) to verify the fabric is healthy.
+Troubleshooting Slingshot is beyond the scope of this document.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
Troubleshooting HPE Slingshot is beyond the scope of this document. See the _HPE Slingshot Troubleshooting Guide_ for troubleshooting information.
```

File: operations/configuration_management/Configure_HSN_NIC_Bonding.md [hunk 15]
```diff
@@ -0,0 +1,418 @@
...
+
+## Additional steps
+
+This procedure performs a one time configuration of the target nodes. The bonded HSN configuration will persist through a reboot of the node but a rebuild of the node
+will wipe it.
+
+In order to persist this configuration through a rebuild of the node, the CFS layer should be added to the CFS configuration used for the NCN Worker nodes.
```
Comment (nrockershousen) [inline operations/configuration_management/Configure_HSN_NIC_Bonding.md]:
```suggestion
In order to persist this configuration through a rebuild of the node, the CFS layer should be added to the CFS configuration used for the NCN worker nodes.
```

**Comment by spillerc-hpe** [comment]
@nrockershousen I incorporated your recommendations and fixed that indentation.

**Review by trad511 — APPROVED** [review]

**Review by shunr-hpe — APPROVED** [review]

**Pull request merged**